core/src/format.rs

```diff
@@ -19,19 +19,21 @@ impl Display for FormatError {
 
 /// Format a Nickel file being read from `input`, writing the result to `output`.
 pub fn format(mut input: impl Read, mut output: impl Write) -> Result<(), FormatError> {
-    let topiary_config =
-        topiary::Configuration::parse_default_configuration().map_err(FormatError)?;
-    let language = topiary::SupportedLanguage::Nickel.to_language(&topiary_config);
     let grammar = tree_sitter_nickel::language().into();
     let query = TopiaryQuery::new(&grammar, topiary_queries::nickel()).map_err(FormatError)?;
-    topiary::formatter(
+    let language = Language {
+        name: "nickel".to_owned(),
+        query,
+        grammar,
+        indent: None,
+    };
+
+    formatter(
         &mut input,
         &mut output,
-        &query,
-        language,
-        &grammar,
-        topiary::Operation::Format {
-            skip_idempotence: true,
+        &language,
+        Operation::Format {
+            skip_idempotence: false,
```

Comment:
Is there guidance for when idempotence should be checked in a downstream usecase like here? I think I originally put it to "skip" because I assumed it's mostly a test for topiary itself.

Reply:
This is my understanding too. I wouldn't like to make users unable to format their codebase because Topiary happens to fail the idempotence checks. On the other hand, it's useful data for debugging Topiary. Maybe we should enable it with the debug profile?

Reply:
Heeeej!

The idempotency check takes up but a tiny fraction of Topiary's runtime, which is why we generally enable it by default.

But yes, in theory it should be fine to disable it.

Reply:
I think I don't mind the extra time, but I fear user can't format their code if they run into an idempotency bug. Or maybe we can catch the error and just emit a warning kindly asking to report the issue?